Update README.md (#1)
viam-labs/onnx-cpu: README.md

# `onnx-cpu` modular resource

> This module allows you to run ONNX models with the Viam Vision Service on Android.

Reviewer: ```suggestion
This module allows you to run ONNX models with the Viam `mlmodel` vision service on Android.
```
Makes it seem like there is only one vision service otherwise

> This module allows you to run ONNX models with the Viam `mlmodel` vision service on Android.

Author: ```suggestion
This module allows you to deploy ONNX models which you can then use with the Viam `mlmodel` vision service on Android.
```

## Requirements

Before configuring your ML model, you must [create a machine](https://docs.viam.com/fleet/machines/#add-a-new-machine).

## Build and run

To use this module, follow these instructions to [add a module from the Viam Registry](https://docs.viam.com/registry/configure/#add-a-modular-resource-from-the-viam-registry) and select the `viam-labs:mlmodel:onnx-cpu` model from the [`onnx-cpu` module](https://github.com/viam-labs/onnx-cpu).

> ## Configure your `onnx-cpu` sensor

Reviewer: So wait-- this is a sensor? That is not clear from intro. Or is it an mlmodel service? seems like service but confusing name

> ## Configure your `onnx-cpu` sensor

Author: ```suggestion
## Configure your `onnx-cpu` ML model
```

Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com/).
Click on the **Services** subtab and click **Create service**.
Select the `mlmodel` type, then select the `onnx-cpu:timesyncsensor` model.
Click **Add module**, then enter a name for your ML model and click **Create**.

On the new component panel, copy and paste the following attribute template into your ML model’s **Attributes** box:

```json
{
  "model_path": "/path/to/onnx_file/detector_googlenet.onnx",
  "label_path": "/path/to/labels.txt"
}
```

> [!NOTE]
> For more information, see [Configure a Machine](https://docs.viam.com/manage/configuration/).

### Attributes

> The following attributes are available for the `viam-labs:mlmodel:onnx-cpu` sensor:

Reviewer: ```suggestion
The following attributes are available for the `viam-labs:mlmodel:onnx-cpu` ML model service:
```

| Name    | Type   | Inclusion    | Description |
| ------- | ------ | ------------ | ----------- |
| `model_path` | string | **Required** | The full path to the ONNX file. |
| `label_path` | string | Optional | The full path to the names of the category labels. |

### Example configuration

```json
    {
      "name": "onnx",
      "type": "mlmodel",
      "model": "viam-labs:mlmodel:onnx-cpu",
      "attributes": {
        "model_path": "/path/to/onnx_file/detector_googlenet.onnx"
        "label_path": "/path/to/labels.txt"
      }
    }
```

> ### Configure a vision service

Reviewer: Transition sentence explaining why configuring vision service could be good beforehand

Author: Good point! Added!

> Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com/).

Author: ```suggestion
The module allows you deploy an ONNX ML model with the ML model service it provides. Viam's vision service allows you to make use of the deployed detection or classification model.

Navigate to the **Config** tab of your machine's page in [the Viam app](https://app.viam.com/).
```

> Click on the **Services** subtab and click **Create service**.

Author: ```suggestion
The module allows you deploy an ONNX ML model with the ML model service it provides. Viam's vision service allows you to make use of the deployed detection or classification model.

Click on the **Services** subtab and click **Create service**.
```

Select the `mlmodel` type, then select the `onnx-cpu:timesyncsensor` model.
Click **Add module**, then enter a name for your ML model and click **Create**.

On the new component panel, copy and paste the following attribute template into your ML model’s **Attributes** box:

```json
{
  "model_path": "/path/to/onnx_file/detector_googlenet.onnx",
  "label_path": "/path/to/labels.txt"
}
```

> [!NOTE]
> For more information, see [Configure a Machine](https://docs.viam.com/manage/configuration/).

### Attributes

The following attributes are available for the `viam-labs:mlmodel:onnx-cpu` ML model service:

| Name    | Type   | Inclusion    | Description |
| ------- | ------ | ------------ | ----------- |
| `model_path` | string | **Required** | The full path to the ONNX file. |
| `label_path` | string | Optional | The full path to the names of the category labels. |

### Example configuration

```json
    {
      "name": "onnx",
      "type": "mlmodel",
      "model": "viam-labs:mlmodel:onnx-cpu",
      "attributes": {
        "model_path": "/path/to/onnx_file/detector_googlenet.onnx"
        "label_path": "/path/to/labels.txt"
      }
    }
```

### Configure a vision service

> The module allows you deploy an ONNX ML model with the ML model service it provides. Viam's vision service allows you to make use of the deployed detection or classification model.

Reviewer: ```suggestion
The module allows you to deploy an ONNX ML model with the ML model service it provides. Viam's vision service allows you to make use of the deployed detection or classification model.
```
